pam-u2f.c

```diff
@@ -278,6 +282,38 @@ int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
     }
   }
 
+  // Determine the lock file path for touch request notifications
+  if (!cfg->lock_file) {
+    buf = NULL;
+
+    char *run_folder = "/var/run/user/";
+    uid_t uid = getuid();
```

Comment:
I think that getting the effective user ID would be more appropriate, this also allows us to reuse the value we get a few lines above.

Reply:
`geteuid()` returned `0` for me, while `getuid()` returned the expected `1000`, which is my user ID. I can double check, but I think `getuid()` is more preferable.